### PR TITLE
HPCC-22690 Change JlibSemTestStress::testSimple to use realistic timings and prevent to getting stuck indefinitely.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Change upper time limit from '2 * time' to 'time + <one minute>'

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [X] Send notifications about my Pull Request position in Smoketest queue.
- [X] Test my draft Pull Request.

## Testing:

Tested manually

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith  Is this what we are talked earlier? (It is finished on my Ubuntu VM without any error, Earlier it is failed 20 times from 20 runs)

File: testing/unittests/jlibtests.cpp [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 #include "unittests.hpp"
 
+#define ONE_MINUTE 60000 // msec
```
Comment (jakesmith) [inline testing/unittests/jlibtests.cpp]: minor: nicer as a typed static const e.g.
```
static const unsigned oneMinute = 60000; // msec
```

File: testing/unittests/jlibtests.cpp [hunk 2]
```diff
@@ -129,7 +131,7 @@ class JlibSemTestStress : public CppUnit::TestFixture
             VStringBuffer errMsg("Semaphore stalled (%s:%d)", sanitizeSourceFile(__FILE__), __LINE__);
             CPPUNIT_FAIL(errMsg.str());
         }
-        testTimedElapsed(sem, 5, 1000);
+        testTimedElapsed(sem, 20, 1000);
```
Comment (jakesmith) [inline testing/unittests/jlibtests.cpp]: not sure why 5 change to 20 here?
(and NB: not in other places, like line 78)

**Comment by jakesmith** [comment]
@AttilaVamos - a couple of comments, but yes think a constant max time change is good.

There's also a similar potential problem with testTimedAvailable though.
i.e. it could take >=5 ms.

I think I'd change it to:

```
    void testTimedAvailable(Semaphore & sem)
    {
        //Shouldn't cause a reschedule, definitely shouldn't wait for 100s
        if (!sem.wait(100))
            ASSERT(false);
    }
```

since that's what it's actually testing afaics.

**Comment by AttilaVamos** [comment]
@jakesmith updated, please re-check.

File: testing/unittests/jlibtests.cpp [hunk 3]
```diff
@@ -43,11 +45,9 @@ class JlibSemTest : public CppUnit::TestFixture
 
     void testTimedAvailable(Semaphore & sem)
     {
-        unsigned now = msTick();
-        sem.wait(100);
-        unsigned taken = msTick() - now;
         //Shouldn't cause a reschedule, definitely shouldn't wait for 100s
-        ASSERT(taken < 5);
+        if(!sem.wait(100))
+            ASSERT(false)
```
Comment (jakesmith) [inline testing/unittests/jlibtests.cpp]: missing ';' (seems to compile anyway, but for consistency)

**Comment by jakesmith** [comment]
@richardkchapman - Not sure important, but I think could be targeted to 7.6.0 ?

@AttilaVamos - 1 minor comment.

File: testing/unittests/jlibtests.cpp (hunk 3 above)
Comment (AttilaVamos) [reply testing/unittests/jlibtests.cpp]: Changed

**Comment by AttilaVamos** [comment]
@jakesmith updated.

**Comment by jakesmith** [comment]
@AttilaVamos - looks good, please squash - and I think can be 7.6.0 too.

**Comment by AttilaVamos** [comment]
@jakesmith squashed and re-targeted.

**Comment by jakesmith** [comment]
Looks good.

@richardkchapman - ready to merge

**Pull request merged**